lib/gamnit/bmfont.nit

```diff
@@ -0,0 +1,422 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Parse Angel Code BMFont format and draw text
+#
+# The BMFont format supports packed textures, varying advance per character and
+# even kernings. It can be generated with a number of tools, inluding:
+# * BMFont, free software Windows app, http://www.angelcode.com/products/bmfont/
+# * Littera, a web app, http://kvazars.com/littera/
+#
+# Format reference: http://www.angelcode.com/products/bmfont/doc/file_format.html
+module bmfont
+
+private import dom
+
+import font
+
+# BMFont description, parsed with `Text::parse_bmfont` or loaded as a `BMFontAsset`
+#
+# This class flattens all the `info` and `common` data.
+class BMFont
+
+	# ---
+	# info part
+	#
+	# How the font was generated.
+
+	# Name of the source true type font
+	var face: Text
+
+	# Size of the source true type font
+	var size: Int
+
+	# Is the font bold?
+	var bold: Bool
+
+	# Is the font italic?
+	var italic: Bool
+
+	# Does the font uses the Unicode charset?
+	var unicode: Bool
```

Comment:
The value comes from the font description file, if `unicode != 1` then we should look at `charset`. But since `charset` is not implemented, `unicode` it is not very useful as of now.

However, it is important because the charset is used to identify characters by their code point! In a previous version there was an assert `unicode == true`, however, I removed it since the tool that I use does not set it correctly in the generated file.

So yeah, it's half implemented but it would be worth completing when we have a better understanding of what different tools can generate.